Common/Data/Consolidators/TradeBarConsolidatorBase.cs

```diff
@@ -66,11 +67,17 @@ protected TradeBarConsolidatorBase(Func<DateTime, CalendarInfo> func)
         }
 
         /// <summary>
-        /// Gets a copy of the current 'workingBar'.
+        /// Creates a consolidator to produce a new 'TradeBar' representing the last count pieces of data or the period, whichever comes first
         /// </summary>
-        public TradeBar WorkingBar
+        /// <param name="pyfuncobj">Python function object that defines the start time of a consolidated data</param>
+        protected TradeBarConsolidatorBase(PyObject pyfuncobj)
+            : base(pyfuncobj)
         {
-            get { return (TradeBar) WorkingData; }
         }
+
+        /// <summary>
+        /// Gets a copy of the current 'workingBar'.
+        /// </summary>
+        public TradeBar WorkingBar => (TradeBar) WorkingData;
```

Comment:
Why add this? It hides the base's property. _Also_ -- typically, if you're going to hide a base class property you should be explicit by using the `new` keyword:

``` csharp
public new TradeBar WorkingBar => (TradeBar)WorkingData;
```

I'm not a fan of this change. If the goal is to provide strongly typed data, then why not do it using generics? `PeriodCountConsolidatorBase<T, TradeBar>` exposes the correct types, so this definition could be pushed all the up (or down depending on how you look at it) the type heirarchy so that _everyone_ gets this behavior instead of just this class _and also_ we won't need to hide properties which can get yucky since behavior becomes dependent on the static (compile-time) type of an object. I've seen some pretty gnarly bugs from such semantics.

Feel free to reach out to discuss further.

Reply:
@AlexCatarino ?

Reply:
The `WorkingBar` was already present, I believe he just refactored into an expression-bodied property, right @AlexCatarino ?

Reply:
Right, @Martin-Molinero.